MeNext/Playing/MainController.cs

```diff
@@ -0,0 +1,384 @@
+		/// </summary>
+		/// <param name="song">Song.</param>
+		public void RequestRemoveFromPlayNext(ISong song)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Atomic operation which is functionally equivalent to:
+		/// <code>
+		/// RequestRemoveFromPlayNext(song);
+		/// RequestAddToPlayNext(song, index);
+		/// </code>
+		/// </summary>
+		/// <param name="song">Song.</param>
+		/// <param name="index">Index.</param>
+		public void MoveWithinPlayNext(ISong song, int index)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Requests that the current user thumbs up the song.
+		/// </summary>
+		/// <param name="song">Song.</param>
+		public void RequestThumbUp(ISong song)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Requests that the current user thumbs down the song.
+		/// </summary>
+		/// <param name="song">Song.</param>
+		public void RequestThumbDown(ISong song)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Requests that the current user's thumb up or down information is cleared for the song.	
+		/// </summary>
+		/// <param name="song">Song.</param>
+		public void RequestThumbNeutral(ISong song)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Requests that the current event be given permissions
+		/// </summary>
+		public void RequestEventPermissions(/* TODO: permissions structure */)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Requests that a user be given permissions.
+		/// </summary>
+		/// <param name="username">Username.</param>
+		public void RequestUserPermissions(string username /* TODO: permissions structure */)
+		{
+			Debug.Assert(this.InEvent);
+		}
+
+		/// <summary>
+		/// Begins polling the server for the latest info
+		/// </summary>
+		private void _StartPolling()
+		{
+			Debug.Assert(this.InEvent);
+			var message = new StartPollMessage(EventSlug, UserKey);
+			MessagingCenter.Send(message, "StartPollMessage");
+		}
+
+		/// <summary>
+		/// Stops polling the server
+		/// </summary>
+		private void _StopPolling()
+		{
+			Debug.Assert(this.InEvent);
+			var message = new StopPollMessage();
+			MessagingCenter.Send(message, "StopPollMessage");
+		}
+
+		/// <summary>
+		/// Subscribes to the polling status.
+		/// </summary>
+		private void SubscribePollingStatus()
+		{
+			MessagingCenter.Subscribe<StatusMessage>(this, "StatusMessage", message =>
+			{
+				if (!this.InEvent)
+						return;
+
+				// Print the debug text
+				if (message.TestingText != null)
+					Debug.WriteLine("DEBUG TEXT: " + message.TestingText);
+
+				// If the event is ended, we should leave and return to the homepage.
+				if (!message.EventActive)
+					LeaveEvent();
+
+				// TODO: Update stuff based on message
+
+				// Update music playing if we are the host
+				if (IsHost)
+				{
+					playController.UpdateActualPlaying(message);
+				}
+
+				// Update UI
+                this.SomethingChanged();
```

Comment:
OK, doesn't look like SomethingChanged has a param? Maybe pass message to it now?